src/components/scenario/ScenarioPanel.tsx

```diff
@@ -0,0 +1,269 @@
+  useScrollTrigger,
+} from '@mui/material';
+import { useTranslation } from 'next-i18next';
+import { Sliders } from 'react-bootstrap-icons';
+
+import type {
+  GetInstanceGoalOutcomeQuery,
+  GetInstanceGoalOutcomeQueryVariables,
+} from '@/common/__generated__/graphql';
+import { activeGoalVar, scenarioEditorDrawerOpenVar, yearRangeVar } from '@/common/cache';
+import { useInstance } from '@/common/instance';
+import { useSiteWithSetter } from '@/context/site';
+
+import GoalSelector from '../general/GoalSelector';
+import NormalizationWidget from '../general/NormalizationWidget';
+import YearRangeSelector from '../general/YearRangeSelector';
+import ScenarioOutcome from './ScenarioOutcome';
+import ScenarioSelector from './ScenarioSelector';
+
+export const GET_INSTANCE_GOAL_OUTCOME = gql`
+  query GetInstanceGoalOutcome($goal: ID!) {
+    instance {
+      id
+      goals(id: $goal) {
+        values {
+          year
+          goal
+          actual
+          isForecast
+          isInterpolated
+        }
+        unit {
+          htmlShort
+        }
+      }
+    }
+  }
+`;
+
+/**
+ * Determines if the scenario panel is fixed and if its minimized.
+ * The minimized version shows the scenario selector and edit scenario button,
+ * hiding the title and year selectors.
+ *
+ * The panel sticks upon scroll past the initial relative position. We display the
+ * full panel when scrolling up, and the minimized version when scrolling down.
+ *
+ * Also return the initial height of the relative panel to support rendering a
+ * placeholder when the panel is fixed and prevent content jumping.
+ */
+function useIsPanelStuck(ref: React.RefObject<HTMLDivElement | null>) {
+  const theme = useTheme();
+  const [position, setPosition] = useState({ top: 0, initialHeight: 0 });
+  const isScrollingDown = useScrollTrigger();
+  const isMobile = useMediaQuery(theme.breakpoints.down('md'));
+  const hasScrolledPastPanelStart = useScrollTrigger({
+    disableHysteresis: true,
+    threshold: position.top,
+  });
+  const hasScrolledPastPanelEnd = useScrollTrigger({
+    disableHysteresis: true,
+    threshold: position.top + position.initialHeight,
+  });
+
+  const isPanelFixed = !isMobile && hasScrolledPastPanelStart;
+  const isPanelMini = isScrollingDown && isPanelFixed && hasScrolledPastPanelEnd;
+
+  function handleChangePosition() {
+    if (ref.current) {
+      setPosition((position) => ({
+        ...position,
+        top: ref.current?.getBoundingClientRect().top ?? position.top + window.pageYOffset,
+      }));
+    }
+  }
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      setPosition({
+        top: ref.current.getBoundingClientRect().top + window.pageYOffset,
+        initialHeight: ref.current.getBoundingClientRect().height,
+      });
+    }
+
+    window.addEventListener('resize', handleChangePosition);
+
+    return () => {
+      window.removeEventListener('resize', handleChangePosition);
+    };
+  }, [ref]);
```

Comment:
@woodwoerk this is a new warning, I have been struggling with the exhaustive dependency arrays myself recently (function dependencies triggering unnecessary renders). Do you think there is a workaround to fix this?